config.h

```diff
@@ -1,7 +1,13 @@
+#if defined(__APPLE__)
+#include <TargetConditionals.h>
+#endif
+
 #if defined(WIN32) || defined(_WIN32)
 #include "config_windows_x86_64.h"
 #elif defined(__gnu_linux__)
 #include "config_linux_x86_64.h"
+#elif defined(TARGET_OS_MAC) && TARGET_OS_MAC
```

Comment:
Both `defined(TARGET_OS_MAC)` and TARGET_OS_MAC are not required, all non-existent preprocessor identifiers are expanded to 0 in preprocessor conditionals. If this is defined as always 1 when true (rather than just to an empty value — and according to what I can find, it is indeed defined as 1), just a `TARGET_OS_MAC` check is enough.